### DISCUSSION TITLE
how to traverse nn.Module (not params) and print submodule

### DISCUSSION THREAD

**Discussion by luweizheng** [google/flax#2264] [Q&A] [answered]
Hi all,

I want to traverse a module's submodule and see what's in this model. 

[Model Surgery](https://flax.readthedocs.io/en/latest/howtos/model_surgery.html) and [tabulate](https://flax.readthedocs.io/en/latest/_autosummary/flax.linen.tabulate.html#flax.linen.tabulate) can get info about params.

However, I want to get not only submodule with params but also activation layers like ReLU.

In torch, we can:

```
for module in model.children():
    
```

How to iterate a flax model defined by  `nn.Module`

**Comment by marcvanzee** [comment, marked as answer]
Module tabulation currently doesn't show information about activation functions, and neither does the variable dictionary. Iterating over module like in Pytorch currently isn't possible either.

There is probably a way for us to support this if we keep track of the initializers in module.py, but it will add some complexity and it currently isn't on our rader. Do you want to open an issue for this? If there is more interest from the community we can start prioritizing it.
